docs/platforms/Swift-Package-Manager.md

```diff
@@ -0,0 +1,645 @@
+    3. If your package must be linked explicitly `static` or `dynamic` ([not recommended](https://developer.apple.com/documentation/packagedescription/product/library(name:type:targets:))), update the [Product](https://developer.apple.com/documentation/packagedescription/product) to define the type
+    ```swift
+    products: [
+        .library(name: "plugin-name-ios", type: .static, targets: ["plugin_name_ios"])
+    ],
+    ```
+    4. Make any other customizations - see https://developer.apple.com/documentation/packagedescription for more info on how to write a Package.swift.
+    5. If you add additional targets to your Package.swift, try to name them uniquely. If your target name conflicts with another target from another package, this can cause issues that may require manual intervention to be able to use your plugin.
+
+14. Update your `plugin_name_ios.podspec` to point to new paths.
+```diff
+- s.source_files = 'Classes/**/*.{h,m}'
++ s.source_files = 'plugin_name_ios/Sources/plugin_name_ios/**/*.{h,m}'
+
+- s.public_header_files = 'Classes/**/*.h'
++ s.public_header_files = 'plugin_name_ios/Sources/plugin_name_ios/include/**/*.h'
+
+- s.module_map = 'Classes/cocoapods_plugin_name_ios.modulemap'
++ s.module_map = 'plugin_name_ios/Sources/plugin_name_ios/include/cocoapods_plugin_name_ios.modulemap'
+
+- s.resource_bundles = {'plugin_name_ios_privacy' => ['Resources/PrivacyInfo.xcprivacy']}
++ s.resource_bundles = {'plugin_name_ios_privacy' => ['plugin_name_ios/Sources/plugin_name_ios/PrivacyInfo.xcprivacy']}
+```
+
+15. Update getting of resources from bundle to use `SWIFTPM_MODULE_BUNDLE`
+```objc
+#if SWIFT_PACKAGE
+   NSBundle *bundle = SWIFTPM_MODULE_BUNDLE;
+ #else
+   NSBundle *bundle = [NSBundle bundleForClass:[self class]];
+ #endif
+ NSURL *imageURL = [bundle URLForResource:@"image" withExtension:@"jpg"];
+```
+  * Note: `SWIFTPM_MODULE_BUNDLE` will only work if there are actual resources (either [defined in the Package.swift](https://developer.apple.com/documentation/xcode/bundling-resources-with-a-swift-package#Explicitly-declare-or-exclude-resources) or [automatically included by Xcode](https://developer.apple.com/documentation/xcode/bundling-resources-with-a-swift-package#:~:text=Xcode%20detects%20common%20resource%20types%20for%20Apple%20platforms%20and%20treats%20them%20as%20a%20resource%20automatically)). Otherwise, it will fail.
+
+16. If your `plugin_name_ios/Sources/plugin_name_ios/include` directory only contains a `.gitkeep`, you'll want update your `.gitignore` to include the following:
+```gitignore
+!.gitkeep
+```
+
+Then run `flutter pub publish --dry-run` to ensure the `include` directory will be published.
+
+17. Verify plugin still works with CocoaPods
+    1. Disable Swift Package Manager
+      ```
+      flutter config --no-enable-swift-package-manager
+      ```
+    2. Run `flutter run` with the example app and ensure it builds and runs
```

Comment:
I wonder if we should recommend doing podspec checks like

```
pod lib lint packages/packages/image_picker/image_picker_ios/ios/image_picker_ios.podspec --configuration=Debug --skip-tests --use-modular-headers --use-libraries
```

and 
```
pod lib lint packages/packages/image_picker/image_picker_ios/ios/image_picker_ios.podspec --configuration=Debug --skip-tests --use-modular-headers
```

We do these in our CI

Reply:
Good idea, added!